cpp/src/io/json/write_json.cu

```diff
@@ -254,38 +183,208 @@ std::unique_ptr<column> struct_to_strings(table_view const& strings_columns,
+                                             string_view const element_narep,
+                                             //  string_view const list_narep,
+                                             rmm::cuda_stream_view stream,
+                                             rmm::mr::device_memory_resource* mr)
+{
+  CUDF_FUNC_RANGE();
+  // create string_views of the list elements, and the list separators and list prefix/suffix.
+  // then concatenates them all together.
+  // gather offset of first string_view of each row as offsets for output string column.
+  auto const offsets          = lists_strings.offsets();
+  auto const strings_children = lists_strings.get_sliced_child(stream);
+  auto const null_count       = lists_strings.null_count();
+  auto const num_lists        = lists_strings.size();
+  auto const num_strings      = strings_children.size();
+  auto const num_offsets      = offsets.size();
+
+  rmm::device_uvector<size_type> d_strview_offsets(num_offsets, stream);
+  auto num_str_views_per_list = cudf::detail::make_counting_transform_iterator(
+    0, [offsets = offsets.begin<size_type>(), num_offsets] __device__(size_type idx) {
+      if (idx + 1 >= num_offsets) return 0;
+      auto const length = offsets[idx + 1] - offsets[idx];
+      return length == 0 ? 2 : (2 + length + length - 1);
+      // TODO nullmask unused for now.
+    });
+  thrust::exclusive_scan(rmm::exec_policy(stream),
+                         num_str_views_per_list,
+                         num_str_views_per_list + num_offsets,
+                         d_strview_offsets.begin());
+  auto const string_views_total = d_strview_offsets.back_element(stream);
+
+  using str_pair = thrust::pair<const char*, size_type>;
+  rmm::device_uvector<str_pair> d_strviews(string_views_total, stream);
+  // scatter null_list and list_prefix, list_suffix
+  auto col_device_view = cudf::column_device_view::create(lists_strings.parent(), stream);
+  thrust::for_each(
+    rmm::exec_policy(stream),
+    thrust::make_counting_iterator<size_type>(0),
+    thrust::make_counting_iterator<size_type>(num_lists),
+    [col = *col_device_view,
+     list_prefix,
+     list_suffix,
+     d_strview_counts = d_strview_offsets.begin(),
+     d_strviews       = d_strviews.begin()] __device__(auto idx) {
+      if (col.is_null(idx)) {
+        d_strviews[d_strview_counts[idx]] = str_pair{nullptr, 0};
+        d_strviews[d_strview_counts[idx] + 1] =
+          str_pair{nullptr, 0};  // TODO could skip this to reduce memory usage.
+      } else {
+        // [ ]
+        d_strviews[d_strview_counts[idx]] = str_pair{list_prefix.data(), list_prefix.size_bytes()};
+        d_strviews[d_strview_counts[idx + 1] - 1] =
+          str_pair{list_suffix.data(), list_suffix.size_bytes()};
+      }
+    });
+
+  // scatter string and separator
+  auto labels = cudf::lists::detail::generate_labels(lists_strings, num_strings, stream, mr);
```

Comment:
```suggestion
  auto labels = cudf::lists::detail::generate_labels(lists_strings, num_strings, stream, rmm::mr::get_current_device_resource());
```
`labels` seems not to be returned by the function

Reply:
updated.